### PR TITLE
Get themes from external repo and build tokens locally

### PR DESCRIPTION
**Pull request by KerimRahman**
Open for discussion.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,13 @@
...
+    "build:cdsTokens": "npm run clone:themes && npm run transform && npm run resolve-tokens",
+    "transform": "npm run transform-light && npm run transform-dark",
+    "transform-light": "npx token-transformer themes tokens/_cds-light/tokens.json core/cds-core,cds-light/theme-cds-light,components/accordion,components/alert,components/background,components/badge,components/breadcrumb,components/button,components/card,components/dropdown,components/form,components/grid,components/header,components/modal,components/pagination,components/spacing,components/tab,components/table,components/toast,components/tooltip,components/type core/cds-core,cds-light/theme-cds-light",
+    "transform-dark": "npx token-transformer themes tokens/_cds-dark/tokens.json core/cds-core,cds-dark/theme-cds-dark,components/accordion,components/alert,components/background,components/badge,components/breadcrumb,components/button,components/card,components/dropdown,components/form,components/grid,components/header,components/modal,components/pagination,components/spacing,components/tab,components/table,components/toast,components/tooltip,components/type core/cds-core,cds-dark/theme-cds-dark",
```
Comment (kidroca) [inline package.json]: Why do we have to list all these items here

```
core/cds-core,cds-light/theme-cds-light,components/accordion,components/alert,components/background...
```

Can they come from config? Or at least something like `components/*` ?

**Review by kidroca — APPROVED** [review]
Looks good overall 

One thing that seems to remain is to fetch the tokens from Figma repo at the build step 
Right now they have to be manually copied to the project right?

Can we continue with a PR for the functionality that fetches the tokens from Figma repo

File: package.json (hunk 1 above)
Comment (KerimRahman) [reply package.json]: I don't think the transformer supports `components/*`. At least, I haven't found that in the docs. Also, that's how it's done in the `cds-token` repo though.

File: build.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
+const StyleDictionaryPackage = require('style-dictionary');
+
+const { fileHeader, formattedVariables } = StyleDictionaryPackage.formatHelpers;
+const { kebabCase, isPx, transformShadow } = require('./utils');
```
Comment (kidroca) [inline build.js]: There are unused items here

File: package.json (hunk 1 above)
Comment (kidroca) [reply package.json]: I've checked out `token-transformer` and the example there just uses `components` and not listing every component folder, can we try that?

File: package.json [hunk 3]
```diff
@@ -31,7 +31,13 @@
...
     "build:storybook": "lerna run build-storybook",
     "build:storybook:react": "cd packages/components/react-web && yarn build-storybook",
-    "build:storybook:wc": "cd packages/components/web-components && yarn build-storybook"
+    "build:storybook:wc": "cd packages/components/web-components && yarn build-storybook",
+    "clone:themes": "git clone --filter=blob:none https://github.com/compassion-global-experience/cds-tokens.git && cp -r cds-tokens/themes/ ./themes/ && rm -rf cds-tokens",
+    "build:cdsTokens": "npm run clone:themes && npm run transform && npm run resolve-tokens",
+    "transform": "npm run transform-light && npm run transform-dark",
+    "transform-light": "npx token-transformer themes tokens/_cds-light/tokens.json core/cds-core,cds-light/theme-cds-light,components/accordion,components/alert,components/background,components/badge,components/breadcrumb,components/button,components/card,components/dropdown,components/form,components/grid,components/header,components/modal,components/pagination,components/spacing,components/tab,components/table,components/toast,components/tooltip,components/type core/cds-core,cds-light/theme-cds-light",
```
Comment (kidroca) [inline package.json]: Can we move these to a separate shell script so it's easy to document everything in the script - why we do stuff, what are the parameters etc...

We should also delete anything unused in the end, or at least add it to gitignore

**Review by kidroca — CHANGES_REQUESTED** [review]
Took another look at the changes and noticed they are added on the root `package.json` and produce results at the root of the monorepo 

This should instead be moved to a separate project in `packages` 

Added a few more notes 

Can you update the PR description with what is expected to happen
I've tried the build command everything runs successfully but I see this at the end:

```
Processing: [_cds-light]

scss
No properties for _cds-light.css. File not created.
...
```
_(same for \_cds-dark)_

It seems no css (variables?) is being output, and the tokens are empty 
`tokens/_cds-light/tokens.json` has this content `{}`

File: package.json [hunk 4]
```diff
@@ -15,7 +15,8 @@
     "packages/elements",
     "packages/examples/*",
     "packages/themes/*",
-    "packages/site"
+    "packages/site",
+    "packages/cds-css"
```
Comment (kidroca) [inline package.json]: The workspace should be just `packages/css` or `packages/css-core`

File: packages/cds-css/package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "@compassion-gds/cds-css",
```
Comment (kidroca) [inline packages/cds-css/package.json]: The name `@compassion-gds/cds-css` is a bit of tautology - CDS is compassion-gds (Compassion Design System) it should be `@compassion-gds/css` or `@compassion-gds/css-core`

**Review by kidroca — CHANGES_REQUESTED** [review]
Looks good 

Can you put the `themes` folder to `.gitignore` 
Similar to how the `dist` folder in other packages is ignored 

The end goal for this package would be to have a `dist` folder with a few css files (e.g. variables-light/dark.css and a bunch of theme-{name}.css)

For the current PR it would be enough to just ignore the `themes` folder and update package names

**Review by kidroca — APPROVED** [review]
LGTM

**Pull request merged**